### PR TITLE
fix(es/minifier): Don't inline conditionally initialized vars

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue:**

 - Closes https://github.com/swc-project/swc/issues/6750.
 - Closes https://github.com/swc-project/swc/issues/6780.

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

File: crates/swc_ecma_minifier/src/program_data.rs [hunk 1]
```diff
@@ -643,7 +643,8 @@ impl ProgramData {
             if ctx.is_op_assign {
                 e.usage_count += 1;
             } else if is_first {
-                if e.ref_count == 1
+                if e.declared
```
Comment (kdy1) [inline crates/swc_ecma_minifier/src/program_data.rs:646]: @Austaras It seems like this is required, but this change causes too many regression.
Any idea?

Comment (Austaras) [reply crates/swc_ecma_minifier/src/program_data.rs:646]: Being dealing with IRL issue lately. Would (hopefully) come back later next week.

Comment (kdy1) [reply crates/swc_ecma_minifier/src/program_data.rs:646]: Sorry to hear that, and no worries. It's not urgent.

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

File: crates/swc_ecma_minifier/tests/full/feedback-mapbox/785-e1932cc99ac3bb67/output.js [hunk 2]
```diff
@@ -1 +1 @@
...
...ect.keys(t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(t);e&&(r=r.filter(function(e){return Object.getOwnPropertyDescriptor(t,e).enumerable})),n.push.apply(n,r)}return n}n4.propTypes=n2,n4.defaultProps={captureScroll:!1,captureDrag:!1,captureClick:!1,captureDoubleClick:!1,capturePointerMove:!1};var n3=Object.assign({},no,{redraw:p.func.isRequired,style:p.object});function n8(t){var e=na(t),n=e.context,r=e.containerRef,i=n.viewport,o=n.isDragging,s=function(t){for(var e=1;e<arguments.length;e++){var n=null!=arguments[e]?arguments[e]:{};e%2?n5(Object(n),!0).forEach(function(e){a(t,e,n[e])}):Object.getOwnPropertyDescriptors?Object.defineProperties(t,Object.getOwnPropertyDescriptors(n)):n5(Object(n)).forEach(function(e){Object.defineProperty(t,e,Object.getOwnPropertyDescriptor(n,e))})}return t}({position:"absolute",left:0,top:0},t.style);return h.createElement("svg",{width:i.width,height:i.height,ref:r,style:s},t.redraw({width:i.width,height:i.height,isDragging:o,project:i.project,unproject:i.unproject}))}n8.propTypes=n3,n8.defaultProps={captureScroll:!1,captureDrag:!1,captureClick:!1,captureDoubleClick:!1,capturePointerMove:!1},tP()&&tP().setRTLTextPlugin}}]);
```
Comment (kdy1) [inline crates/swc_ecma_minifier/tests/full/feedback-mapbox/785-e1932cc99ac3bb67/output.js:1]: Maybe I should remove these tests...

**Review by jridgewell — DISMISSED** [review]
Can you give the new snapshot output of https://github.com/swc-project/swc/issues/6750?

**Comment by kdy1** [comment]
It's

```js
let current_component;
try {
    !function(component) {
        let parent = current_component;
        current_component = component, parent.m();
    }({
        m () {
            console.log("call m()");
        }
    });
} catch (e) {
    console.log('PASS');
}
```

**Comment by kdy1** [comment]
I'll add a snapshot test

**Pull request merged**